test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -62,24 +62,38 @@ var _ = SIGDescribe("GracefulNodeShutdown [Serial] [NodeFeature:GracefulNodeShut
 	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
 
 	ginkgo.BeforeEach(func() {
-		if _, err := exec.LookPath("systemd-run"); err == nil {
-			if version, verr := exec.Command("systemd-run", "--version").Output(); verr == nil {
-				// sample output from $ systemd-run --version
-				// systemd 245 (245.4-4ubuntu3.13)
-				re := regexp.MustCompile(`systemd (\d+)`)
-				if match := re.FindSubmatch(version); len(match) > 1 {
-					systemdVersion, err := strconv.Atoi(string(match[1]))
-					if err != nil {
-						framework.Logf("failed to parse systemd version with error %v, 'systemd-run --version' output was [%s]", err, version)
-					} else {
-						// See comments in issue 107043, this is a known problem for a long time that this feature does not work on older systemd
-						// https://github.com/kubernetes/kubernetes/issues/107043#issuecomment-997546598
-						if systemdVersion < 245 {
-							e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on an old version of systemd : %d", systemdVersion)
-						}
-					}
-				}
-			}
+		if _, err := exec.LookPath("systemd-run"); err != nil {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: %v", err)
+		}
+		version, err := exec.Command("systemd-run", "--version").Output()
+		if err != nil {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: %v", err)
+		}
+
+		// sample output from $ systemd-run --version
+		// systemd 245 (245.4-4ubuntu3.13)
+		re := regexp.MustCompile(`systemd (\d+)`)
+		match := re.FindSubmatch(version)
+		if len(match) <= 1 {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on a non-systemd system: version %s", version)
+		}
+
+		systemdVersion, err := strconv.Atoi(string(match[1]))
+		if err != nil {
+			e2eskipper.Skipf("skipping failed to parse systemd version with error %v, 'systemd-run --version' output was [%s]", err, version)
+		}
+
+		// See comments in issue 107043, this is a known problem for a long time that this feature does not work on older systemd
+		// https://github.com/kubernetes/kubernetes/issues/107043#issuecomment-997546598
+		if systemdVersion < 245 {
+			e2eskipper.Skipf("skipping GracefulNodeShutdown tests as we are running on an old version of systemd : %d", systemdVersion)
+		}
+
+		framework.Logf("Running systemd version %d", systemdVersion)
+
+		err = checkInhibit()
```

Comment:
Skip when dbus it is not working

Reply:
My main concern is will this end up skipping the tests most of the time?

It would be nice to test this in a eventually loop just so we try to make sure this is ready before we skip.

Reply:
Do we know why Dbus is either not running or is somewhat broken?

We recently removed the test restarting Dbus, causing some, if not most, of the systemd services issues.

The systemd-inhibit would only ever fall if it cannot connect to Dbus (including socket activation failure) or the connection timed out.

I wonder what the root cause might be, especially if something we do is causing problems with Dbus.

Reply:
> My main concern is will this end up skipping the tests most of the time?
>
> It would be nice to test this in a eventually loop just so we try to make sure this is ready before we skip.

Done

Reply:
> Do we know why Dbus is either not running or is somewhat broken?
> 
> We recently removed the test restarting Dbus, causing some, if not most, of the systemd services issues.
> 
> The systemd-inhibit would only ever fall if it cannot connect to Dbus (including socket activation failure) or the connection timed out.
> 
> I wonder what the root cause might be, especially if something we do is causing problems with Dbus.

https://github.com/kubernetes/test-infra/issues/29722

Maybe the underlying environment has changed as a result of the migration to community clusters? I'm not sure, and I'm still diagnosing the cause.

Reply:
@wzshiming, I see. It might well be.

Could the sizing of the nodes or allotted resources be an issue here? I see a lot of complaints about tests running into CPU limits. That said, I don't know how significant this is here.

The limit updates (two most recent ones):

- https://github.com/kubernetes/kubernetes/issues/118491
- https://github.com/kubernetes/kubernetes/issues/67621

Not sure if the above has any significance.

However, I am not sure how and why Dbus would be affected. Does it crash? Is it not started? I am not sure about Google's COS, but at least on Fedora CoreOS and Ubuntu, it's a centrepiece for systemd, so a lot of things would break - the most obvious (aside from other services like the systemd-inhibit) would be SSH access and sudo, which would become terribly slow if not inaccessible.

I looked at the build logs - but there is little there in terms of the OS-level information that is being captured. I am not sure where to find OS-level logs if at all possible, to get these.

Should we bump the systemd log level too? Thoughts?